runtime/src/alpha.rs

```diff
@@ -29,20 +29,65 @@ use tokio_stream::wrappers::ReceiverStream;
 use tokio_util::sync::CancellationToken;
 use tonic::{transport::Server, Request, Response, Status};
 
-use crate::args::args;
 use crate::print_version;
 
-// uses custom macro instead of clap to reduce dependency weight
-args! {
-    pub struct Args {
-        // The port to open the gRPC control layer on.
-        // The address to expose for the service is given in the StartRequest.
-        "--port" => pub port: u16,
+#[derive(Default)]
+struct Args {
+    /// Enable compatibility with beta platform
+    beta: bool,
+    /// Alpha (required): Port to open gRPC server on
+    port: Option<u16>,
+    /// Beta (required): Address to bind the gRPC server to
+    address: Option<SocketAddr>,
+}
+
+impl Args {
+    // uses simple arg parsing logic instead of clap to reduce dependency weight
+    fn parse() -> anyhow::Result<Self> {
+        let mut args = Self::default();
+
+        // The first argument is the path of the executable
+        let mut args_iter = std::env::args().skip(1);
+
+        while let Some(arg) = args_iter.next() {
+            match arg.as_str() {
+                "--beta" => {
+                    args.beta = true;
+                }
+                "--port" => {
+                    let port = args_iter
+                        .next()
+                        .context("missing port value")?
+                        .parse()
+                        .context("invalid port value")?;
+                    args.port = Some(port);
+                }
+                "--address" => {
+                    let address = args_iter
+                        .next()
+                        .context("missing address value")?
+                        .parse()
+                        .context("invalid address value")?;
+                    args.address = Some(address);
+                }
+                _ => {}
```

Comment:
Maybe we add some message about invalid args here?

Reply:
We don't want old runtimes to freak out if they recieve new args that they don't support yet.

Reply:
Ah, that's a good point.